### DISCUSSION TITLE
API to delete and move issues and to ban users

### DISCUSSION THREAD

**Discussion by KOLANICH** [community/community#24782] [Apps, API and Webhooks]
<p>I have implemented an Action that automatically closes issues not following the template. But it is incomplete. I actually want to automatically move them into a garbage repo untill they are fixed, and completely remove them if they  are not fixed, and man the users who repeatedly create the issues not following the template. It seems tat the API for all of this is missing.</p>

**Comment by nethgato** [comment, marked as answer]
<p>Hi <a class="mention" href="/kolanich">@kolanich</a> :wave:</p>
<p>So there is a GraphQL mutation to transfer issues between repositories:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/graphql/reference/mutations#transferissue" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/graphql/reference/mutations#transferissue" target="_blank" rel="noopener">Mutations - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/graphql/reference/mutations#transferissue" target="_blank" rel="noopener">//docs.github.com/en/graphql/reference/mutations</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Similarly for deleting issues, you can do this via GraphQL here:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/graphql/reference/mutations#deleteissue" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/graphql/reference/mutations#deleteissue" target="_blank" rel="noopener">Mutations - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/graphql/reference/mutations#deleteissue" target="_blank" rel="noopener">//docs.github.com/en/graphql/reference/mutations</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>You can block users, via:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/rest/reference/users#block-a-user" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/rest/reference/users#block-a-user" target="_blank" rel="noopener">Users - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/rest/reference/users#block-a-user" target="_blank" rel="noopener">//docs.github.com/en/rest/reference/users</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>…being mindful that this blocks users from your personal account, and not at the repository level.</p>
<p>That said, if any of the above doesn’t fully solve for your use case, I would highly recommend submitting your suggestions directly to our PMs via this feedback form:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://support-assets.github.com/assets/favicon-3dcd2c977a91b2e3f396b300c5ff0d572295ca632ddb7ba1adcacc1ea68dd5c0.ico" class="site-icon" width="16" height="16">
      <a href="https://support.github.com/contact/feedback" target="_blank" rel="noopener">support.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://support.github.com/contact/feedback" target="_blank" rel="noopener">Share feedback - GitHub Support</a></h3>

<p><a href="https://support.github.com/contact/feedback" target="_blank" rel="noopener">//support.github.com/contact/feedback</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I hope this is helpful, but of course, if you have any other questions beyond what you’ve submitted, let us know! :bow:</p>

**Comment by KOLANICH** [comment]
<p>Thanks you. About GraphQL I have already figured that out, though I find GQL API for that a bit inconvenient to use because it involves specific request for each mutation rather than incrementally built endpoint URI.</p>
